middleware/stations.js

```diff
@@ -4,24 +4,18 @@ function unenvelope(envelope) {
   return envelope['data']
 }
 
-function fetchStations(callback) {
-
-  fetch('http://pomiary.gdmel.pl/rest/stations')
+function fetchStations() {
+  return fetch('http://pomiary.gdmel.pl/rest/stations')
     .then((res) => res.json())
-    .then((json) => {
-      // console.log(json);
+    .then((json) => unenvelope(json))
+    .then((result) => {
+      let list = []
 
-      return json;
-    }).then((json) => {
-      const data = unenvelope(json);
-      return data
-    }).then((result) => {
-      // console.log(result);
       for(item of result) {
```

Comment:
Iterować <szczególnie jeśli używasz fora na generatorach> zawsze lepiej jest funkcyjnie. Tutaj pewnie wystarczy użyć result.forEach()

Reply:
O chuj jak szanuję! Ogólnie to całą funkcję można skrócić do <pre>result.map(station => station<b>s</b>ToModel(station))</pre>, jeżeli result jest tablicą ;)

Przyczepie się do tego 's', bo to jest false advertising, ta funkcja konwertuje stacj**ę** (jedną) do modelu